docs/connect-blockchain/cosmos.mdx

```diff
@@ -0,0 +1,377 @@
+---
+title: Integrate Web3Auth with the Cosmos Blockchain
+sidebar_label: Cosmos
+image: "content-hub/guides/banners/cosmos.png"
+displayed_sidebar: docs
+keywords: [cosmos, web3auth, authentication, blockchain]
+description: "Integrate Web3Auth with the Cosmos Blockchain | Documentation - Web3Auth"
+---
+
+import GetUserInfoSnippet from "@site/src/common/docs/_get-userinfo.mdx";
+
+---
+
+While using the Web3Auth Web SDK for a non EVM chain like [Cosmos](https://cosmos.network/) you get a standard provider from which you can get the
+private key of the user. Using this private key, you can use the corresponding libraries of the blockchain to make blockchain calls like getting
+user's `account`, `chain ID`, fetch `balance`, `send transaction`, etc. We have highlighted a few methods here for getting you started quickly on
+that.
+
+:::note
+
+This reference is for the `Web`, however, you can use StarkNet on `Android`, `iOS`, `React Native`, `Flutter`, & `Unity` as well. Please follow our
+reference for [Ethereum](/connect-blockchain/ethereum/), and similarly use Cosmos libraries that support the platforms to use the private key and make
+blockchain calls accordingly.
+
+:::
+
+## Installation
+
+```bash npm2yarn
+npm install --save @web3auth/no-modal @web3auth/openlogin-adapter @web3auth/base @cosmjs/stargate @cosmjs/proto-signing
+```
+
+## Initializing Provider
+
+---
+
+```js
+import { Web3Auth } from "@web3auth/modal";
+import { OpenloginAdapter } from "@web3auth/openlogin-adapter";
+
+const web3auth = new Web3Auth({
+  clientId: "YOUR_WEB3AUTH_CLIENT_ID", // get it from Web3Auth Dashboard
+  web3AuthNetwork: "cyan",
+  chainConfig: {
+    chainNamespace: "other", // for all non EVM and SOLANA chains, use "other"
+    rpcTarget: "https://rpc.sentry-02.theta-testnet.polypore.xyz",
+    displayName: "Cosmos",
+    blockExplorer: "https://explorer.theta-testnet.polypore.xyz",
+    ticker: "ATOM",
+    tickerName: "Cosmos",
+  },
+});
+
+// "other" is supported through @web3auth/openlogin-adapter package.
+const openloginAdapter = new OpenloginAdapter({
+  adapterSettings: {
+    uxMode: "popup",
+  },
+});
+web3auth.configureAdapter(openloginAdapter);
+
+await web3auth.initModal();
+
+const web3authProvider = web3auth.connect(); // web3auth.provider
+```
+
+## Get User Info
+
+---
+
+<GetUserInfoSnippet />
+
+## Get ChainID
+
+---
+
+Once a user logs in, the Web3Auth SDK returns a provider. Since Web3Auth doesn't have a native provider for Cosmos, we need to directly use the
+private key to make the RPC calls.
+
+### CosmJS Libraries
+
+`@cosmjs/stargate` A client library for the Cosmos SDK 0.40+. Provides a high level client for querying, signing and broadcasting. This is an npm
+library that provides a JavaScript/TypeScript interface to interact with the Cosmos SDK blockchain via the Tendermint RPC interface.
+`npm install --save @cosmjs/stargate`
```

Comment:
Looks like this is the previous version. Please push the recent one we went through after standup.

Reply:
Oh, I forgot to push those changes. Just did that.